src/coreComponents/mesh/EmbeddedSurfaceNodeManager.cpp

```diff
@@ -0,0 +1,416 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file EmbeddedSurfaceNodeManager.cpp
+ */
+
+#include "EmbeddedSurfaceNodeManager.hpp"
+#include "FaceManager.hpp"
+#include "EdgeManager.hpp"
+#include "ToElementRelation.hpp"
+#include "BufferOps.hpp"
+#include "common/TimingMacros.hpp"
+#include "mpiCommunications/MpiWrapper.hpp"
+#include "ElementRegionManager.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+
+EmbeddedSurfaceNodeManager::EmbeddedSurfaceNodeManager( string const & name,
+                                                        Group * const parent ):
+  ObjectManagerBase( name, parent ),
+  m_referencePosition( 0, 3 )
+{
+  registerWrapper( viewKeyStruct::referencePositionString(), &m_referencePosition );
+  this->registerWrapper( viewKeyStruct::edgeListString(), &m_toEdgesRelation );
+  this->registerWrapper( viewKeyStruct::elementRegionListString(), &elementRegionList() );
+  this->registerWrapper( viewKeyStruct::elementSubRegionListString(), &elementSubRegionList() );
+  this->registerWrapper( viewKeyStruct::elementListString(), &elementList() );
+  this->registerWrapper( viewKeyStruct::parentEdgeGlobalIndexString(), &m_parentEdgeGlobalIndex );
+}
+
+
+EmbeddedSurfaceNodeManager::~EmbeddedSurfaceNodeManager()
+{}
+
+
+void EmbeddedSurfaceNodeManager::resize( localIndex const newSize )
+{
+  m_toEdgesRelation.resize( newSize, 2 * getEdgeMapOverallocation() );
+  m_toElements.m_toElementRegion.resize( newSize, 2 * getElemMapOverAllocation() );
+  m_toElements.m_toElementSubRegion.resize( newSize, 2 * getElemMapOverAllocation() );
+  m_toElements.m_toElementIndex.resize( newSize, 2 * getElemMapOverAllocation() );
+  ObjectManagerBase::resize( newSize );
+}
+
+
+void EmbeddedSurfaceNodeManager::setEdgeMaps( EdgeManager const & embSurfEdgeManager )
```

Comment:
This is probably the exact same stuff from `NodeManager` right?

Reply:
@TotoGaz will refactor most of the mesh and move all these functions that create the maps so I will leave it for now.